### DISCUSSION TITLE
For custom types

### DISCUSSION THREAD

**Discussion by huanyu-li** [mirumee/ariadne#804] [Q&A] [answered]
Hi,

I just have a question regarding how to create a type for my data rather than basic scalar types (int, string, boolean, float and ID). For instance, I may need to represent double datatype to datetme datatype.

Thank you in advance.
Best,
Huanyu Li

**Comment by rafalp** [comment]
I'm not sure what you mean? Are you looking to define custom types in GraphQL schema? Have you seen [this document](https://ariadnegraphql.org/docs/resolvers)?

**Reply by huanyu-li** [reply]
Thanks for your reply.
I'm building a GraphQL application over some data sources where there are date time type data. I'm not sure how to define a GraphQL schema to represent date time data.

**Comment by rafalp** [comment, marked as answer]
I see, you want custom scalar then. Here is documentation how to make one with examples including datetime:

https://ariadnegraphql.org/docs/scalars

**Reply by khanakia** [reply]
The documentation is not quite clear after I define the scalar how do register it ?

```
JSON = ScalarType("JSON", serializer=lambda v: v, value_parser=lambda v: v)
```
Now where to register the `JSON` field ?

**Reply by rafalp** [reply]
You pass it to `make_executable_schema`.
